src/curl.c

```diff
@@ -75,8 +75,13 @@ curl_multi_process(CURLM *multi, curl_socket_t sockfd, int events)
 	CURLMsg *msg;
 	int msgs_left;
 	while ((msg = curl_multi_info_read(multi, &msgs_left))) {
-		if (msg->msg != CURLMSG_DONE)
+		/* See https://curl.se/libcurl/c/curl_multi_info_read.html */
+		if (msg && (msg->msg != CURLMSG_DONE)) {
+			CURL *e = msg->easy_handle;
+			curl_multi_remove_handle(multi, msg->easy_handle);
+			curl_easy_cleanup(e);
 			continue;
+		}
```

Comment:
From the commit message:

> Documentation for this function [1] describes
how messages with status `CURLMSG_DONE` must be handled: one need call
`curl_multi_remove_handle(multi_handle, e)` and then
`curl_easy_cleanup(e)`.

But the  msg->msg != CURLMSG_DONE. This is not a problem now because so the code is not executed:

> When msg is CURLMSG_DONE, the message identifies a transfer that is done, and then result contains the return code for the easy handle that just completed.

> At this point, there are no other msg types defined.

Reply:
Right, commit has been removed.